cpp/libs/gst_nmos_sender_plugin/utils.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <string>
+#include <glib.h>
+#include <nlohmann/json.hpp>
+
+typedef struct NodeFields
+{
+    gchar* id;
+    gchar* configuration_location;
+} NodeFields;
+
+typedef struct DeviceFields
+{
+    gchar* id;
+    gchar* label;
+    gchar* description;
+} DeviceFields;
+
+typedef struct MediaFields
+{
+    gint width;
+    gint height;
+    gint frame_rate_num;
+    gint frame_rate_density;
+    gchar* sampling;
+    gchar* structure;
+} MediaFields;
+
+typedef struct NetworkFields
+{
+    gchar* source_address;
+    gchar* interface_name;
+    gchar* destination_address;
+    gchar* destination_port;
+} NetworkFields;
+
+typedef struct ConfigFields
+{
+    NodeFields node;
+    DeviceFields device;
+    MediaFields media;
+    gchar* sender_id;
+    gchar* sender_label;
+    gchar* sender_description;
+    NetworkFields network;
+} ConfigFields;
+
+ConfigFields create_default_config_fields();
+
+nlohmann::json create_node_config(const ConfigFields& config);
+nlohmann::json create_device_config(const ConfigFields& config);
+nlohmann::json create_sender_config(const ConfigFields& config);
+
+const std::string get_node_id(const char* node_configuration_location);
```

Comment:
`const` in a return by copy is useless. Please remove all of these.
It would make sense if you were returning a reference, e.g. `const std::string&`.